.github/workflows/ci.yml

```diff
@@ -44,26 +42,31 @@ jobs:
       test-jobs: ${{ steps.project-jobs.outputs.test-jobs }}
       report-jobs: ${{ steps.project-jobs.outputs.report-jobs }}
     steps:
-      - uses: 'actions/checkout@v4'
-        name: 'Checkout'
+      - name: Setup - calculate checkout depth
```

Comment:
The job performance boost, checkout takes ~4 seconds instead of 40 without any loss of functionality.